File: 剑指Offer/替换空格.java
public class Solution {
    public String replaceSpace(StringBuffer str) {
        //System.out.println(str.length());
        for (int i = 0;i < str.length();i++){
            if (str.charAt(i) == ' '){
                str.replace(i,i + 1,"%20");
            }
        }
        return str.toString();
    }
}

File: 剑指Offer/C++/替换空格.cpp
class Solution {
public:
    void replaceSpace(char *str,int length) {
        int i = 0,blank = 0;
        while (str[i] != '\0'){
            if (str[i++] == ' '){
                blank++;
            }
        }
        int originLength = i;
        int newLength = originLength + blank * 2;
        while (originLength >= 0 && originLength <= newLength){
            if (str[originLength] == ' '){
                str[newLength--] = '0';
                str[newLength--] = '2';
                str[newLength--] = '%';
                originLength--;
            }else{
                str[newLength--] = str[originLength--];
            }
        }
    }
};